java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -73,4 +74,30 @@ public interface StreamBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> xtrim(String key, StreamTrimOptions options);
+
+    /**
+     * Reads entries from the given streams.
+     *
+     * @see <a href="https://redis.io/commands/xread/">redis.io</a> for details.
+     * @param keysAndIds - A <code>Map</code> of keys and entry ids to read from. The <code>
+     *     Map</code> is composed of a stream's key and the id of the entry after which the stream
+     *     will be read.
+     * @return A <code>Map</code> of a stream key to an array of entries in the matching <code>key
+     *     </code>. The entries are in an <code>[id, fields[]]</code> format.
+     */
+    CompletableFuture<Map<String, Map<String, String[]>>> xread(Map<String, String> keysAndIds);
```

Comment:
see my comment above about data type - you need to update response type and docs